packages/owl-bot/src/bin/commands/enqueue-copy-tasks.ts

```diff
@@ -0,0 +1,149 @@
+        demand: true,
+      })
+      .option('app-id', {
+        describe: 'GitHub AppID',
+        type: 'number',
+        demand: true,
+      })
+      .option('installation', {
+        describe: 'installation ID for GitHub app',
+        type: 'number',
+        demand: true,
+      })
+      .option('repo', {
+        describe: 'repository to monitor for changes',
+        type: 'string',
+        default: 'googleapis/googleapis-gen',
+      })
+      .option('git-path', {
+        describe: 'path on disk of repo to monitor for changes',
+        type: 'string',
+        demand: true,
+      })
+      .option('project', {
+        describe: 'gcloud project',
+        type: 'string',
+        default: 'repo-automation-bots',
+      })
+      .option('firestore-project', {
+        describe: 'project used for firestore database',
+        type: 'string',
+        default: 'repo-automation-bots-metrics',
+      })
+      .option('queue', {
+        describe: 'queue to publish PR update jobs to',
+        type: 'string',
+        default: 'projects/repo-automation-bots/topics/owlbot-prs',
+      })
+      .option('trigger', {
+        describe:
+          'enqueue copy job trigger (if this should be run in background)',
+        type: 'string',
+      });
+  },
+  async handler(argv) {
+    const privateKey = await readFileAsync(argv['pem-path'], 'utf8');
+    if (!privateKey) throw Error('pem-path or private-key must be provided');
+    if (argv.trigger) {
+      await triggerEnqueueCopyJobs({
+        privateKey,
+        appId: argv['app-id'],
+        installation: argv.installation,
+        project: argv.project,
+        firestoreProject: argv['firestore-project'],
+        queue: argv.queue,
+        trigger: argv.trigger,
+      });
+    } else {
+      const token = await getGitHubShortLivedAccessToken(
+        privateKey,
+        argv['app-id'],
+        argv.installation
+      );
+      const octokit = await getAuthenticatedOctokit(token.token);
+      admin.initializeApp({
+        credential: admin.credential.applicationDefault(),
+        projectId: argv['firestore-project'],
+      });
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const db = admin.firestore();
+      const configStore = new FirestoreConfigsStore(db!);
+
+      const shas = await getLastNCommits(argv.repo, octokit, 1);
+      console.info(shas);
+      // TODO: stop hardcoding this once we've tested the system:
+      const sha = 'b32470c59b90b1bb22c29d285474c19870d6f3f1';
+      const files = await getFilesModifiedBySha(argv['git-path'], sha);
+      logger.info(`found ${files.length} files changed`);
+      const repos = await configStore.findReposAffectedByFileChanges(files);
+      for (const repo of repos) {
+        let messageId = await configStore.findPubsubMessageIdForCopyTask(
+          repo,
+          sha
+        );
+        if (messageId) {
+          logger.info(`${repo} has copy job ${messageId}`);
```

Comment:
Let's include the decision we're making in this log line.  For example,
`${repo} already has copy job ${messageId}, so I won't create another one.`